app/test_main.py

```diff
@@ -1 +1,58 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+URL = "http//:www.google.com/"
+
+
+@pytest.fixture()
+def mocked_valid_google_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mocked_func:
+        yield mocked_func
+
+
+@pytest.fixture()
+def mocked_has_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mocked_func:
+        yield mocked_func
+
+
+def test_valid_connection_and_url_exist(
+        mocked_valid_google_url: mock,
+        mocked_has_connection: mock) -> None:
```

Comment:
You should move the last bracket to the new line as well )


```suggestion
def test_valid_connection_and_url_exist(
        mocked_valid_google_url: mock,
        mocked_has_connection: mock
) -> None:
```

Reply:
Fixed)